sql/core/src/main/scala/org/apache/spark/sql/GroupedData.scala

```diff
@@ -273,6 +280,60 @@ class GroupedData protected[sql](
   def sum(colNames: String*): DataFrame = {
     aggregateNumericColumns(colNames : _*)(Sum)
   }
+
+  /**
+    * (Scala-specific) Pivots a column of the current [[DataFrame]] and preform the specified
+    * aggregation.
+    * {{{
+    *   // Compute the sum of earnings for each year by course with each course as a separate column
+    *   df.groupBy($"year").pivot($"course", "dotNET", "Java").agg(sum($"earnings"))
+    *   // Or without specifying column values
+    *   df.groupBy($"year").pivot($"course").agg(sum($"earnings"))
+    * }}}
```

Comment:
There is a note below in the description of the values parameter
>If values are not provided the method with do an immediate call to .distinct() on the pivot column.
Do we need to duplicate/move that note to the method description?

Right now if there is too many distinct values we probably get OOM. Obvious solve is to have some configurable maximum above which we give an error. Should I try to add that now?